src/components/App/App.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import { Route, BrowserRouter, Switch } from "react-router-dom";
+import NavBar from "../NavBar/NavBar";
+import VideoContext from "../../providers/VideoContext";
+import Home from "../../pages/Home/Home";
+import LogIn from "../../pages/Login/LogIn";
+import ReproductorPlayer from "../../pages/ReproductorPlayer/ReproductorPlayer";
+
+const App = () => {
+  require('dotenv').config()
+  const [response, setResponse] = useState({});
+  const [videoMetaInfo, setVideoMetaInfo] = useState([]);
+  const [changingg, changinggSetChangingg] = useState(false);
+  const [favoriteVideosMapUser, setFavoriteVideosMapUser] = useState(new Map());
+  const [favoritesFlag, setFavoritesFlag] = useState(false);
+  const [favoritesVideosArrayUser, setFavoritesVideosArrayUser] = useState([]);
+
+  const [videoPropperties, setSelectedVideoPropperties] = useState({
+    videoName: "",
+    videoId: "",
+    views: "",
+    likes: "",
+    dislikes: "",
+    published: "",
+    imageVideo: "",
+    buttonText: "Add to favorites"
+  });
+  const [userSession, setUserSession] = useState({
+    loggedIn: false,
+    user: "",
+    pass: "",
+    email: ""
+  });
+
+  return (
+    <div className="App">
+      <BrowserRouter>
+        <VideoContext.Provider
```

Comment:
As a good practice you can move this component to a different file named ```VideoProvider``` for example. Also instead of exposing the setter for each state is better if you create a handler and expose that handler. Finally I saw that you are using multiple states, maybe your code could look cleaner if you use ```useReducer``` instead of the state hook.